### PR TITLE
Removed deprecated code from hashicorp provider

### PR DESCRIPTION
**Pull request by Prab-27**
related: #44559 

Removed deprecated code from hashicorp provider

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: providers/src/airflow/providers/hashicorp/hooks/vault.py [hunk 1]
```diff
@@ -148,22 +146,7 @@ def __init__(
...
-            elif self.connection.extra_dejson.get("role_id"):
-                role_id = self.connection.extra_dejson.get("role_id")
-                warnings.warn(
-                    """The usage of role_id in connection extra for AppRole authentication has been
-                    deprecated. Please use connection login.""",
-                    AirflowProviderDeprecationWarning,
-                    stacklevel=2,
-                )
-            elif self.connection.login:
+            if self.connection.login:
```
Comment (shahar1) [inline providers/src/airflow/providers/hashicorp/hooks/vault.py]: Maybe join with the `if` above? (`if auth_type == "approle" and self.connection.login:`)

**Review by shahar1 — COMMENTED** [review]
Good job!
I wonder why no unit test are failing - could you please check it out?

**Comment by Prab-27** [comment]
Done

File: providers/src/airflow/providers/hashicorp/hooks/vault.py [hunk 2]
```diff
@@ -385,7 +368,6 @@ def get_connection_form_widgets(cls) -> dict[str, Any]:
                 description="Must be 1 or 2.",
                 default=DEFAULT_KV_ENGINE_VERSION,
             ),
-            "role_id": StringField(lazy_gettext("Role ID (deprecated)"), widget=BS3TextFieldWidget()),
```
Comment (Prab-27) [inline providers/src/airflow/providers/hashicorp/hooks/vault.py:388]: I'm bit confused about this ,
Could you please review it and let me know if any changes are required.

File: providers/src/airflow/providers/hashicorp/hooks/vault.py (hunk 1 above)
Comment (Prab-27) [reply providers/src/airflow/providers/hashicorp/hooks/vault.py]: Done

File: providers/src/airflow/providers/hashicorp/hooks/vault.py (hunk 2 above)
Comment (shahar1) [reply providers/src/airflow/providers/hashicorp/hooks/vault.py:388]: What are you confused about? Whether we should remove this line?
If you referred to that, I think that it's ok to remove it as it's a part of the deprecated features. If you meant something else, or wrote this comment here by mistake, please let me know.

**Review by shahar1 — APPROVED** [review]

File: providers/src/airflow/providers/hashicorp/hooks/vault.py (hunk 2 above)
Comment (Prab-27) [reply providers/src/airflow/providers/hashicorp/hooks/vault.py:388]: Yes, I was not sure whether to remove it at that time. Now you clarified, so it's all done.

**Pull request merged**